analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -70,28 +77,30 @@ protected override void Initialize(SonarAnalysisContext context)
                             continue;
                         }
 
-                        if (typeSymbol.OriginalDefinition.IsAny(genericTypes))
+                        if (typeSymbol.OriginalDefinition.IsAny(GenericTypes))
```

Comment:
Lines 74-77: I don't like the `continue`. Let's change the `foreach` to this, because the `typeSyntax` is not used for anything else.
```
                    foreach (var typeSymbol in implementedTypes.Value.Select(x => c.SemanticModel.GetSymbolInfo(x.Type).Symbol?.GetSymbolType()).WhereNotNull())

```

Reply:
When looking at it, we can actually significantly improve performance of this rule by doing this instead:
```
var containingType = (INamedTypeSymbol)c.ContainingSymbol;
foreach (var typeSymbol in containingType.Interfaces.Concat(new[] { containingType.BaseType }).WhereNotNull())
```
We can avoid re-querying the symbols from syntax, as we already know the outer symbol.